drivers/ethernet/eth_mcux.c

```diff
@@ -853,7 +853,11 @@ static int eth_rx(struct eth_context *context)
 			ptpTimeData.second--;
 		}
 
-		pkt->timestamp.nanosecond = ptpTimeData.nanosecond;
+		/* The offset added corresponds to the time it takes for the packet 
+		* to transit between the phy and the mac. This constant is board dependant 
+		* but is working accross with both mimxrt1170_evk and imxrt1050_evkb
+		*/
+		pkt->timestamp.nanosecond = ts-840;
```

Comment:
If this is board dependent, then it feels like this should be a Kconfig

Reply:
@lomn, thank you for the PR. What process did you use to determine the value?

Reply:
Yes this is board dependent I should update it... To determine the constant I used two avnu certified devices and a small ethernet cable. I determined the propagation delay through the cable this way. I then connected the board with one of the two devices and measured the pdelay again except this time there was the offset added by the board. you then just add/substract the offset in software. I also took into account the pdelay variations as it oscilate with an amplitude of approximately 10 to 20ns depending on the board and the PLL of the timer.
On the mimxrt1170_evk there is a way better way to do this with the QOS mac but it's not yet supported.

Reply:
This change will glitch around the second transition... to be done properly it must be done on the whole 64bit (or more) absolute time.
Can this be moved to the gptp stack using a board specific define (since usually all board/soc, require mac<-> phy) delay compensation.